### PR TITLE
ecosystem page

### PR DESCRIPTION
**Pull request by 0xFlackoLFG**

### CODE REVIEW

**Review by luzhijing — APPROVED** [review]
LGTM

File: scripts/update.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+# Build TiDB
+echo "updating and building TiDB..."
+cd $GOPATH/src/github.com/pingcap/tidb
+git pull
+
+make
+cp -f ./bin/tidb-server $TIDB_PATH/bin
+cd $TIDB_PATH
+echo "TiDB is built"
+
```
Comment (jeffreys-cat) [inline scripts/update.sh:37]: $TIDB, these should not be here

File: scripts/upload.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+def progress_handler(progress, total):
+    print("{}/{} {:.2f}".format(progress, total, progress/total*100))
+
+# local_file: local file path
+# remote_name: 上传到七牛后保存的文件名
+def upload_to_qiniu(local_file, remote_name, ttl=3600):
+    print('uploading to qiniu', local_file, remote_name, ttl)
+    #构建鉴权对象
+    q = Auth(QINIU_ACCESS_KEY, QINIU_SECRET_KEY)
```
Comment (jeffreys-cat) [inline scripts/upload.py:29]: We don't have QINIU-ACCESS_KEY and QINIU-SECRET_KEY. We are uploading the issue using Alibaba Cloud

File: scripts/merge_by_toc.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+# coding: utf8
+#
+# Generate all-in-one Markdown file for `doc-cn`
+# Tip: 不支持中文文件名
+# readme.md 中的目录引用的md多次（或者md的sub heading)，以第一次出现为主
```
Comment (jeffreys-cat) [inline scripts/merge_by_toc.py:6]: Delete Chinese

**Pull request closed without merging**